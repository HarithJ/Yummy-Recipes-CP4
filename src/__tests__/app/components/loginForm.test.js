import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import LoginForm from '../../../components/auth/login/loginForm.js'

describe("login form component", () => {
  const login = shallow(<LoginForm />)
  const preventDefault = jest.fn();

  it("renders properly", () => {
    expect(shallowToJson(login)).toMatchSnapshot();
  })

  it('it renders state initially', () => {
    expect(login.state().email).toEqual('');
    expect(login.state().password).toEqual('');
  });

  it('has the correct form fields', () => {
   expect(login.find('#inputName')).toHaveLength(1);
   expect(login.find('#inputPassword')).toHaveLength(1);
   expect(login.find('[type="submit"]')).toHaveLength(1);
 });

 it('Form fields update when state changes', () => {
    login.setState({ password: 'Password', email: 'email' });
    expect(login.find('#inputName').props().value).toEqual('email');
    expect(login.find('#inputPassword').props().value).toEqual('Password');
  });

  it('has handleSubmit function', () => {
    expect(login.instance().handleSubmit({ preventDefault }));
  })


  it('has hideAlert function', () => {
    expect(login.instance().hideAlert());
  })
})
