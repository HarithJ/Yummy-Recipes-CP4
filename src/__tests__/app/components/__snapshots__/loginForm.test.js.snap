// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form component renders properly 1`] = `
<div
  className="container-fluid mTop"
>
  <div
    className="row justify-content-center"
  >
    <form
      className="col-sm-6"
      onSubmit={[Function]}
    >
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={500}
        transitionEnter={true}
        transitionEnterTimeout={500}
        transitionLeave={true}
        transitionLeaveTimeout={500}
        transitionName="example"
      />
      <h5
        className="text-secondary"
      >
        Sign in!!!
      </h5>
      <div
        className="form-group"
      >
        <label
          htmlFor="inputName"
        >
          Name
        </label>
        <input
          aria-describedby="emailHelp"
          className="form-control"
          id="inputName"
          name="name"
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="inputPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="inputPassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <div
        className="row justify-content-end"
      >
        <div
          className="col-auto"
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Sign in
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
